src/Parse/ParseObject.php

```diff
@@ -907,6 +907,8 @@ public function _encode()
         foreach ($this->estimatedData as $key => $value) {
             if (is_object($value) && $value instanceof Encodable) {
                 $out[$key] = $value->_encode();
+            } elseif (self::isAssoc($value)) {
+                $out[$key] = $value;
             } elseif (is_array($value)) {
                 $out[$key] = [];
                 foreach ($value as $item) {
```

Comment:
Here is where the issue starts. You can remove the code above and instead modify this foreach as follows:
```php
foreach($value as $itemKey => $item) {
```
As you can see within that **foreach** `$out` has entries appended to it, like `$out[$key][] = $item->_encode();` and `$out[$key][] = $item;`. The issue you're observing is the array's keys (when associative) are being discarded here, and replaced with numeric indexes instead.

If you can modify those two lines mentioned above to use `$itemKey` (or something similar) that will fix the issue, without having to do any additional checking. Just keep in mind that **$key** is already used, but you can use any other key name you like.